be/src/common/configbase.cpp

```diff
@@ -38,6 +38,14 @@ std::map<std::string, std::string>* full_conf_map = nullptr;
 
 Properties props;
 
+// Because changes to the std::string type are not atomic,
+// we introduce a lock to protect mutable string type config item.
+std::mutex mstring_conf_lock;
+
+std::mutex* get_mstring_conf_lock() {
```

Comment:
update config & update cache not in one lock, is it still have concurrency issue?